src/components/UI/MainButton.vue

```diff
@@ -1,5 +1,5 @@
 <template>
-  <button :disabled="disabled" :class="buttonClasses" @click="callback">
+  <button :disabled="disabled" :class="buttonClasses" @click="$emit('clicked')">
```

Comment:
that's exactly what I was thinking about. I just wonder if maybe `click` is not a bit better name because this way your `<MainButton>` would be more similar to native `<button>`? one more thing that would be nice (because `MainButton` pretends to be reusable component) is to emit original native `click` event here. this way:
1. `<MainButton>` would be even more similar to native `<button>`
2. in the future you will be able to e.g. call `preventDefault()` on this native event from parent perspective.

p.s. you can always pass arow function `(event) => $emit('click', event)` to get original event or the original event can be referenced as `$event` if you use inline function (the way it is right now) 😉